CoreDataStackTests/InitializationTests.swift

```
@@ -47,4 +47,94 @@ class CoreDataStackTests: TempDirectoryTestCase {
+        XCTAssertNotNil(stack.privateQueueContext)
+        XCTAssertTrue(fileExists("TestModel.sqlite", directory: .CachesDirectory))
+    }
+    
+    func testInitializationWithStoreURLOverrideInDirectory() {
+        let bundle = NSBundle(forClass: CoreDataStackTests.self)
+        let ex1 = expectationWithDescription("SQLite Callback")
+        removeIfExists("TestModel.sqlite", directory: .CachesDirectory)
+        CoreDataStack.constructSQLiteStack(withModelName: "TestModel", inBundle: bundle, withStoreURL: tempStoreURL, inDirectory: .CachesDirectory) { result in
+            switch result {
+            case .Success(let stack):
+                self.stack = stack
+            case .Failure(let error):
+                print(error)
+                XCTFail()
+            }
+            ex1.fulfill()
+        }
+        
+        waitForExpectationsWithTimeout(10, handler: nil)
+        
+        XCTAssertNotNil(stack.mainQueueContext)
+        XCTAssertNotNil(stack.privateQueueContext)
+        XCTAssertFalse(fileExists("TestModel.sqlite", directory: .CachesDirectory))
+    }
+    
+    private func fileExists(fileName:String,directory:NSSearchPathDirectory) -> Bool {
+        let urls = NSFileManager.defaultManager().URLsForDirectory(directory, inDomains: .UserDomainMask)
+        let fileURL = urls.first?.URLByAppendingPathComponent(fileName)
+        if let filePath = fileURL?.path {
+            return NSFileManager.defaultManager().fileExistsAtPath(filePath)
+        }
+        else {
+            return false
+        }
+    }
+    
+    private func removeIfExists(fileName:String,directory:NSSearchPathDirectory) {
+        let urls = NSFileManager.defaultManager().URLsForDirectory(directory, inDomains: .UserDomainMask)
+        let fileURL = urls.first?.URLByAppendingPathComponent(fileName)
+        if let filePath = fileURL?.path {
+            if NSFileManager.defaultManager().fileExistsAtPath(filePath) {
```

Good point: Don't eat the error; log it so that you can debug your likely failing test. :)